### ISSUE TITLE
Get access to EpollDomainSocketChannel in ServerInterceptor 

### ISSUE THREAD

**Issue by bdnett** [grpc/grpc-java#2646]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?
grpc 1.0.3

### What JVM are you using (`java -version`)?
java version "1.8.0_111"

### Feature Request
We'd like to get access to the EpollDomainSocketChannel to access method peerCredentials() (new in Netty release 4.1.7) from within a gRPC server's BindableService call.

We're using gRPC with Unix Domain Sockets (Netty's EpollServerDomainSocketChannel) on a server to manage a hardware resource.  We allow a single user (but multiple processes) to open the resource more than once, but we need to reserve the resource while that user has it open and not allow any other users access to the device.  Our server needs to be resilient and detect that the client program has potentially crashed and allow the resource to be cleaned up and made available for another user.

We can add a ServerInterceptor but still I haven't figured out to get the Channel.

Can we get access from a ServerCall object to the Channel (understanding that it may be null for an Inprocess server)?  Or other alternatives?

**Comment by carl-mastrangelo** [comment]
Is that API stable?  On the surface, it looks pretty similar to the ssl creds for a connection, so it would probably use the same mechanism to access them (Grpc.TRANSPORT_ATTR_SSL_SESSION, a transport Attribute.Key)

**Comment by ejona86** [comment]
Yeah, grepc calling peerCredentials() and providing those creds in ServerCall.attributes seems reasonable.

**Comment by bdnett** [comment]
An attribute for PeerCredentials should work fine for us.
